src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -0,0 +1,642 @@
+    )
+
+    # TODO: add atom index-based instantiator for species if not present
+
+    hamiltonian_scope = Quantity(
+        type=EnergyContribution,
+        shape=['*'],
+        description="""
+        Reference to the section `EnergyContribution` containing the information
+        of the Hamiltonian term to which the basis set applies.
+        """,
+        a_eln=ELNAnnotation(components='ReferenceEditQuantity'),
+    )
+
+    # ? band_scope or orbital_scope: valence vs core
+
+    def normalize(self, archive, logger):
+        super().normalize(archive, logger)
+        self.name = self.m_def.name
+
+
+class PlaneWaveBasisSet(BasisSet, Mesh):
+    """
+    Basis set over a reciprocal mesh, where each point $k_n$ represents a planar-wave basis function $\frac{1}{\sqrt{\omega}} e^{i k_n r}$.
+    Typically the grid itself is cartesian with only points within a designated sphere considered.
+    The cutoff radius may be defined by a reciprocal length, or more commonly, the equivalent kinetic energy for a free particle.
+
+    * D. J. Singh and L. Nordström, \"Why Planewaves\" in Planewaves, pseudopotentials, and the LAPW method, 2nd ed. New York, NY: Springer, 2006, pp. 24-26.
+    """
+
+    cutoff_energy = Quantity(
+        type=np.float64,
+        unit='joule',
+        description="""
+        Cutoff energy for the plane-wave basis set.
+        The simulation uses plane waves with energies below this cutoff.
+        """,
+    )
+
+    @property  # ? keep, or convert to `Quantity`?
+    def cutoff_radius(self) -> pint.Quantity:
+        """
+        Compute the cutoff radius for the plane-wave basis set, expressed in reciprocal coordinates.
+        """
+        if self.cutoff_energy is None:
+            return None
+        m_e = const.m_e * ureg(const.unit('electron mass'))
+        h = const.h * ureg(const.unit('Planck constant'))
+        return np.sqrt(2 * m_e * self.cutoff_energy) / h
+
+    def normalize(self, archive: EntryArchive, logger: BoundLogger) -> None:
+        super(BasisSet, self).normalize(archive, logger)
+        super(Mesh, self).normalize(archive, logger)
+
+
+class APWPlaneWaveBasisSet(PlaneWaveBasisSet):
+    """
+    A `PlaneWaveBasisSet` specialized to the APW use case.
+    Its main descriptors are defined in terms of the `MuffinTin` regions.
+    """
+
+    cutoff_fractional = Quantity(
+        type=np.float64,
+        shape=[],
+        description="""
+        The spherical cutoff parameter for the interstitial plane waves in the APW family.
+        This cutoff has no units, referring to the product of the smallest muffin-tin radius
+        and the length of the cutoff reciprocal vector ($r_{MT} * |K_{cut}|$).
+        """,
+    )
+
+    def set_cutoff_fractional(
+        self, mt_r_min: pint.Quantity, logger: BoundLogger
+    ) -> None:
+        """
+        Compute the fractional cutoff parameter for the interstitial plane waves in the LAPW family.
```

Comment:
Please, add Args definitions and what this are, e.g., what is `mt_r_min`.

Also, why not being more verbose?